cmd/notation/plugin.go

```diff
@@ -62,3 +108,169 @@ func listPlugins(command *cobra.Command) error {
 	}
 	return tw.Flush()
 }
+
+func installPlugin(command *cobra.Command, args []string, force bool) error {
+	if len(args) != 1 {
+		return errors.New("missing plugin package")
+	}
+
+	pluginPath := args[0]
+
+	var pluginBinary string
+	// if plugin contains a file path split and select the last element in the array
+	if strings.Contains(pluginPath, "/") {
+		pluginBinary = strings.Split(pluginPath, "/")[len(strings.Split(pluginPath, "/"))-1]
+	} else {
+		pluginBinary = pluginPath
+	}
+
+	// get plugin metadata
+	cmd := exec.Command("./"+pluginPath, "get-plugin-metadata")
+
+	output, err := cmd.Output()
+	if err != nil {
+		panic(err)
+	}
+
+	var newPlugin map[string]interface{}
+	err = json.Unmarshal([]byte(output), &newPlugin)
+	if err != nil {
+		return err
+	}
+
+	pluginName := newPlugin["name"].(string)
+	newPluginVersion := newPlugin["version"].(string)
+	newSemVersion, err := semver.NewVersion(newPluginVersion)
+	if err != nil {
+		return err
+	}
+
+	// get plugin directory
+	pluginDir, err := dir.PluginFS().SysPath(pluginName)
+	if err != nil {
+		panic(err)
+	}
+
+	// Check if plugin directory exists
+	_, err = os.Stat(pluginDir)
+
+	// create the directory, if not exist
+	if os.IsNotExist(err) {
+		if err := os.MkdirAll(pluginDir, 0755); err != nil {
+			return err
+		}
+	}
+
+	// Check if plugin exists
+	_, err = os.Stat(pluginDir + "/" + pluginBinary)
+
+	// copy plugin, if not exist
+	if os.IsNotExist(err) {
```

Comment:
Same here. Are we handling the case where `err != nil` AND `os.IsNotExist(err) == false` ?

Reply:
Helper function called `exists`, as well as, error handling for `err != nil`.